torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -709,10 +718,22 @@ Stmt* LoopNest::lowerToStmt(Tensor* t) {
     throw malformed_input("Tensor lowered to zero dimensions");
   }
 
+  const Expr* initializer = t->initializer();
+  if (initializer) {
+    buf_initializers_[t->func_var()] = initializer;
+  }
+  int rdims = f->ndim() - t->ndim();
+  std::vector<const Expr*> indices(t->args().begin(), t->args().end());
+
   for (size_t i = 0; i < f->ndim(); i++) {
     // Going in reverse order: from innermost loop to the outermost
     size_t dim_index = f->ndim() - i - 1;
     body = new For(f->arg(dim_index), new IntImm(0), f->dim(dim_index), body);
+    indices.pop_back();
+    if (initializer && indices.size() == t->ndim()) {
```

Comment:
Do we specify anywhere in which order reduction and normal axis should go?

Reply:
We don't, but:
1. So far we have always set the normal axes on the outside and reduction axis inside.
2. We shouldn't rely on any order in any transformations or they won't interact well with e.g. reorderAxis.

Can you think of a situation where it matters because it might affect #2.